### PR TITLE
add toggle on favorite btn

### PR DESCRIPTION
**Pull request by MStartsev**

### CODE REVIEW

File: src/js/favoritesDB/toggleFavorit.js
Comment (Yuriy-St) [inline src/js/favoritesDB/toggleFavorit.js:1]: this function must be implemented in the exercise-modal.js file

File: src/js/favoritesDB/toggleFavorit.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  const toggleBtn = document.getElementById(toggleClass);
+
+  toggleBtn.addEventListener('click', async event => {
+    const isOK = await toggleFavoriteStatus(exercise);
+    console.log(isOK);
+    const text = !isOK ? 'Add to favorites' : 'Remove from favorites';
+    const iconId = !isOK ? 'icon-heart' : 'icon-trash';
+
+    toggleBtn.innerHTML = `<span id="js-toggle-favorit-text">${text}</span>
+       <svg id="js-toggle-favorit-image" class="btn-icon">
```
Comment (Yuriy-St) [inline src/js/favoritesDB/toggleFavorit.js:16]: do we really need the 'id' properties here?

File: src/js/modal/exercise-modal.js [hunk 2]
```diff
@@ -14,7 +16,7 @@ const buttonBoxRef = modalRef.querySelector('.button-box');
 const MAX_RATING = 5;
 
 const renderModal = exercise => {
-  const { gifUrl, name, rating, isFavorite, description } = exercise;
+  const { gifUrl, name, rating, _id, isFavorite, description } = exercise;
```
Comment (Yuriy-St) [inline src/js/modal/exercise-modal.js:20]: _id not used in the function

File: src/js/modal/exercise-modal.js [hunk 3]
```diff
@@ -128,11 +131,11 @@ const markupGiveRatingBtn = () =>
   });
 
 const markupButton = ({ text, iconId, className = '' }) => `
-  <button type="button" class="js-favorites button ${className}">
+  <button id="js-toggle-favorit" type="button" class="js-favorites button ${className}">
```
Comment (Yuriy-St) [inline src/js/modal/exercise-modal.js]: this is a 'factory' function and could be used for making different buttons, therefore such 'id' is not correct (my fault with js-favorites class as well)

File: src/js/modal/exercise-modal.js [hunk 4]
```diff
@@ -128,11 +131,11 @@ const markupGiveRatingBtn = () =>
...
       <span>${text}</span>
       ${
         iconId
-          ? `<svg class="btn-icon">
+          ? `<svg  class="btn-icon">
```
Comment (Yuriy-St) [inline src/js/modal/exercise-modal.js]: typo

File: src/js/modal/exercise-modal.js [hunk 5]
```diff
@@ -196,11 +199,15 @@ const exercise = {
   burnedCalories: 220,
   time: 3,
   popularity: 8322,
-  isFavorite: true,
+  isFavorite: false,
 };
 
-export const showModal = () => {
-  openModalExercise(exercise);
+export const showModal = async function () {
```
Comment (Yuriy-St) [inline src/js/modal/exercise-modal.js:211]: we must change this logic because this is function was made just for testing purporses

**Review by SofiaProtsiv — APPROVED** [review]

File: src/js/modal/exercise-modal.js (hunk 2 above)
Comment (MStartsev) [reply src/js/modal/exercise-modal.js:20]: remove

**Pull request merged**